packages/react-components/react-table/src/hooks/selectionManager.ts

```diff
@@ -0,0 +1,112 @@
+import { SelectionMode } from './types';
+
+type OnSelectionChangeCallback = (selectedItems: Set<SelectionItemId>) => void;
+
+export interface SelectionManager {
+  toggleSelect(id: SelectionItemId): void;
+  select(id: SelectionItemId): void;
+  deSelect(id: SelectionItemId): void;
+  clearSelection(): void;
+  isSelected(id: SelectionItemId): boolean;
+  toggleSelectAll(itemIds: SelectionItemId[]): void;
+}
+
+export type SelectionItemId = string | number;
+
+export function createSelectionManager(
+  mode: SelectionMode,
+  onSelectionChange: OnSelectionChangeCallback = () => undefined,
+): SelectionManager {
+  const selectedItems = new Set<SelectionItemId>();
+  return mode === 'multiselect'
+    ? createMultipleSelectionManager(selectedItems, onSelectionChange)
+    : createSingleSelectionManager(selectedItems, onSelectionChange);
```

Comment:
It's a matter of preference, but why not:

```suggestion
  const selectedItems = new Set<SelectionItemId>();
  const managerFactory = mode === 'multiselect' ? createMultipleSelectionManager : createSingleSelectionManager;

  return managerFactory(selectedItems, onSelectionChange);
```

Reply:
yeah that is better, I have to embrace the javascript-ness